### DISCUSSION TITLE
Help with scanning a table

### DISCUSSION THREAD

**Discussion by alexander-schranz** [PaddlePaddle/PaddleOCR#13447] [Q&A]
First of all I want to say that paddleocr is a really nice OCR Toolkit. I'm using the paddleocr script to try:

```bash
.venv/bin/paddleocr  --image_dir 933.png --use_angle_cls true --lang en --use_gpu false --structure_version PP-StructureV2 --show_log false --det_db_unclip_ratio 2 --det_db_box_thresh 0.5 --det_db_thresh 0.2
```

I played around with the different threshold but running into some issues here detecting the POS column:

<img width="203" alt="Bildschirmfoto 2024-07-21 um 20 28 01" src="https://github.com/user-attachments/assets/67bd3ad2-77d9-40fe-9df3-b904344cd19b">

and detect single values at end of the:

<img width="250" alt="Bildschirmfoto 2024-07-21 um 20 28 04" src="https://github.com/user-attachments/assets/94abe236-0b72-49a6-8315-73ddb667eb98">

The whole debug output with all default options is:

<details>
<summary>Output log</summary>

[2024/07/21 20:29:59] ppocr DEBUG: Namespace(help='==SUPPRESS==', use_gpu=False, use_xpu=False, use_npu=False, use_mlu=False, ir_optim=True, use_tensorrt=False, min_subgraph_size=15, precision='fp32', gpu_mem=500, gpu_id=0, image_dir='/tmp/933.png', page_num=0, det_algorithm='DB', det_model_dir='/tmp/.paddleocr/whl/det/en/en_PP-OCRv3_det_infer', det_limit_side_len=960, det_limit_type='max', det_box_type='quad', det_db_thresh=0.2, det_db_box_thresh=0.5, det_db_unclip_ratio=2.0, max_batch_size=10, use_dilation=False, det_db_score_mode='fast', det_east_score_thresh=0.8, det_east_cover_thresh=0.1, det_east_nms_thresh=0.2, det_sast_score_thresh=0.5, det_sast_nms_thresh=0.2, det_pse_thresh=0, det_pse_box_thresh=0.85, det_pse_min_area=16, det_pse_scale=1, scales=[8, 16, 32], alpha=1.0, beta=1.0, fourier_degree=5, rec_algorithm='SVTR_LCNet', rec_model_dir='/tmp/.paddleocr/whl/rec/en/en_PP-OCRv4_rec_infer', rec_image_inverse=True, rec_image_shape='3, 48, 320', rec_batch_num=6, max_text_length=25, rec_char_dict_path='/tmp/Documents/Projects/paddle-ocr/.venv/lib/python3.9/site-packages/paddleocr/ppocr/utils/en_dict.txt', use_space_char=True, vis_font_path='./doc/fonts/simfang.ttf', drop_score=0.5, e2e_algorithm='PGNet', e2e_model_dir=None, e2e_limit_side_len=768, e2e_limit_type='max', e2e_pgnet_score_thresh=0.5, e2e_char_dict_path='./ppocr/utils/ic15_dict.txt', e2e_pgnet_valid_set='totaltext', e2e_pgnet_mode='fast', use_angle_cls=False, cls_model_dir='/tmp/.paddleocr/whl/cls/ch_ppocr_mobile_v2.0_cls_infer', cls_image_shape='3, 48, 192', label_list=['0', '180'], cls_batch_num=6, cls_thresh=0.9, enable_mkldnn=False, cpu_threads=10, use_pdserving=False, warmup=False, sr_model_dir=None, sr_image_shape='3, 32, 128', sr_batch_num=1, draw_img_save_dir='./inference_results', save_crop_res=False, crop_res_save_dir='./output', use_mp=False, total_process_num=1, process_id=0, benchmark=False, save_log_path='./log_output/', show_log=True, use_onnx=False, return_word_box=False, output='./output', table_max_len=488, table_algorithm='TableAttn', table_model_dir=None, merge_no_span_structure=True, table_char_dict_path=None, layout_model_dir=None, layout_dict_path=None, layout_score_threshold=0.5, layout_nms_threshold=0.5, kie_algorithm='LayoutXLM', ser_model_dir=None, re_model_dir=None, use_visual_backbone=True, ser_dict_path='../train_data/XFUND/class_list_xfun.txt', ocr_order_method=None, mode='structure', image_orientation=False, layout=True, table=True, ocr=True, recovery=False, use_pdf2docx_api=False, invert=False, binarize=False, alphacolor=(255, 255, 255), lang='en', det=True, rec=True, type='ocr', savefile=False, ocr_version='PP-OCRv4', structure_version='PP-StructureV2')
[2024/07/21 20:29:59] ppocr INFO: **********/~/tmp/933.png**********

</details>

I tried things like disable the `merge_no_span_structure false` and some different values for det_db_thresh, det_db_box_thresh or det_db_unclip_ratio. But not completly understand which options I need adjust here to make the scan better. 

<details>
<summary>
The file I use can be found here:</summary>

![933(1)](https://github.com/user-attachments/assets/4b1fbac5-d836-4dd4-aba0-1bf344667bf0)

</details>

Does somebody may have a hint here how I can improve the scanning of such document?

**Comment by UserWangZz** [comment]
This situation may require you to fine-tune the table recognition model and text detection model

**Reply by alexander-schranz** [reply]
Thx for the response I created now a `training.txt` file like this:

<details>
<summary>training.txt</summary>

```txt
position/1.png	1
position/2.png	2
position/3.png	3
position/4.png	4
position/5.png	5
position/6.png	6
position/7.png	7
position/8.png	8
position/9.png	9
position/21.png	21
position/21a.png	21a
position/22.png	22
position/21.png	21
position/23.png	23
position/24.png	24
position/25.png	25
position/25a.png	25a
position/26.png	26
position/27.png	27
position/28.png	28
position/29.png	29
position/30.png	30
position/31.png	31
position/32.png	32
position/33.png	33
position/34.png	34
position/35.png	35
position/36.png	36
position/37.png	37
position/38.png	38
position/39.png	39
position/40.png	40
position/41.png	41
position/42.png	42
position/43.png	43
position/44.png	44
position/45.png	45
position-letters/A.png	A
position-letters/B.png	B
position-letters/C.png	C
position-letters/D.png	D
position-letters/E.png	E
count-letters/A.png	A
count-letters/B.png	B
count-letters/C.png	C
count-letters/D.png	D
count-letters/E.png	E
labels/anmarkning-anmerkung.png	AnmarkningAnmerkung
labels/antal-anzahl.png	AntalAnzahl
labels/artikel-nr-artikelnr.png	ArtikelnrArtikelnr
labels/benamning.png	Benämning
labels/bezeichnung.png	Bezeichnung
labels/grupp-gruppe.png	GruppGruppe
labels/reservdelskatalog-ersatzteilkatalog.png	ReservdelskatalogErsatzteilkatalog
labels/sida-seite.png	SidaSeite
labels/volvo.png	VOLVO
count/1.png	1
count/2.png	2
count/4.png	4
count/8.png	8
count/-.png	-
count/x.png	X
notes/(248276)265TE.png	(248276)265TE
notes/(942620).png	(942620)
notes/(943717).png	(943717)
notes/(946551)M6SM10X20.png	(946551)M6SM10X20
notes/(955144).png	(955144)
notes/(968234)ALTER1.M10X50.png	(968234)ALTER1.M10X50
notes/(1205939).png	(1205939)
notes/(1205942).png	(1205942)
notes/(1212199).png	(1212199)
notes/(1221902).png	(1221902)
notes/(1229428).png	(1229428)
notes/245T.png	245T
notes/264TE.png	264TE
notes/ALTER1.MBX22.png	ALTER1.MBX22
notes/ALTER2.M8X16.png	ALTER2.M8X16
notes/ALTER2.M10X30.png	ALTER2.M10X30
notes/EXC245T.png	EXC245T
notes/See36•611.png	See36•611
notes/See88•1509.png	See88•1509
SR/2.png	2
SR/3.png	3
SR/4.png	4
```

</details>

and `training.yml` file like this:

<details>
<summary>training.yml</summary>

```txt
Global:
  pretrained_model: ./.paddleocr/whl/det/en/en_PP-OCRv3_det_slim_distill_train/student.pdparams
Optimizer:
  lr:
    name: Cosine
    learning_rate: 0.001 # learning_rate
    warmup_epoch: 2
  regularizer:
    name: 'L2'
    factor: 0

Train:
  dataset:
    name: SimpleDataSet
    data_dir: ./
    label_file_list:
    - ./training.txt
    ratio_list: [1.0] # Sampling ratio, the default value is [1.0]
  loader:
    shuffle: True
    drop_last: False
    batch_size_per_card: 8  # single gpu batch size
    num_workers: 4
```

</details>

But I'm not understanding what I need to execute to actually train the model and use then the training for my command call: https://github.com/PaddlePaddle/PaddleOCR/blob/main/doc/doc_en/finetune_en.md
